src/Optimizer/Transformer/ServerSideRendering.php

```diff
@@ -605,7 +605,7 @@ private function createResponsiveSizer(
         $style = 'padding-top:%s%%'
     ) {
         $padding       = $height->getNumeral() / $width->getNumeral() * 100;
-        $paddingString = (string) round($padding, 4);
+        $paddingString = sprintf('%.6g', $padding);
```

Comment:
Hmm, I think this is still risky, as a number like '3.33333333' will produce 5 decimals instead of 4.

Okay, so here's an algorithm that should work:
```suggestion
        $paddingString = preg_replace('/\.?0*$/', '', sprintf('%.4F', $padding));
```
We're splitting the operation up into two separate steps, as PHP doesn't seem to provide a single function that solves all our requirements. We first round to a maximum of 4 decimals, and then we strip any unneeded trailing zeros and possibly the separator as well.

So for the back and forth, I would have assumed PHP provides a function to solve this issue, but I can't find one it seems.